ob1k-security/src/main/java/com/outbrain/ob1k/security/server/AuthenticationCookieAesEncryptor.java

```diff
@@ -0,0 +1,66 @@
+package com.outbrain.ob1k.security.server;
+
+import com.ning.http.util.Base64;
+import org.apache.commons.lang3.SerializationUtils;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * A cookie encryptor that use 128bit AES encryption
+ */
+public class AuthenticationCookieAesEncryptor implements AuthenticationCookieEncryptor {
+
+  public static final String AES_ALGORITHM = "AES";
+
+  private static ThreadLocal<Cipher> decryptingCipher;
+  private static ThreadLocal<Cipher> encryptingCipher;
+
+  /**
+   * @param key a 128bit key
+   */
+  public AuthenticationCookieAesEncryptor(final byte[] key) throws NoSuchPaddingException,
+    NoSuchAlgorithmException,
+    InvalidKeyException {
+
+    final SecretKey secretKey = new SecretKeySpec(key, AES_ALGORITHM);
+
+    final Cipher decCypher = Cipher.getInstance(AES_ALGORITHM);
+    decCypher.init(Cipher.DECRYPT_MODE, secretKey);
+    decryptingCipher = new ThreadLocal<>();
+    decryptingCipher.set(decCypher);
```

Comment:
that's not good enough. it will inhabit only the thread on which you created that object. use a thread local initializer instead